params.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package wptdashboard
+
+package wptdashboard
+
+import (
+	"net/http"
+	"net/url"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// ParseSHAParam parses and validates the 'sha' param for the request.
+// It returns "latest" by default (and in error cases).
+func ParseSHAParam(r *http.Request) (runSHA string, err error) {
+	// Get the SHA for the run being loaded (the first part of the path.)
+	runSHA = "latest"
+	params, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		return runSHA, err
+	}
+
+	runParam := params.Get("sha")
+	regex := regexp.MustCompile("[0-9a-fA-F]{10}")
+	if regex.MatchString(runParam) {
+		runSHA = runParam
+	}
+	return runSHA, err
```

Comment:
Can we log here? No error will be reported but we're falling back on something other than what was input to the function.

Reply:
It's user-input, so maybe we should just throw an error. Either way - refactor or plumbing needed, so will follow-up. https://github.com/w3c/wptdashboard/issues/283